### PR TITLE
edited flat_map description and removed typo

### PR DESCRIPTION
**Pull request by Smokrow**
The examples in interleave are quite helpful. I just added a reference to this example for flat_map

### CODE REVIEW

File: tensorflow/python/data/ops/dataset_ops.py
Comment (mrry) [inline tensorflow/python/data/ops/dataset_ops.py]: I think this change might confuse readers. The output of `Dataset.flat_map()` is only identical to `Dataset.interleave()` when `cycle_length=1` (which is an unusual configuration, since it doesn't actually interleave any inputs!). I'm not sure it's worth introducing that concept to readers here.

Additional nits:
* By PEP 257, the first line of a docstring should be a one-line summary; and by the style guide, it's subject to the 80-character limit, so we can't add much more content at this exact location.
* We use an extended form of Markdown in docstrings, and symbol names should be enclosed in backticks ("`") to create hyperlinks for them.

Comment (Smokrow) [reply tensorflow/python/data/ops/dataset_ops.py]: Alright. 
I created a quick example and replaced the "identical" with "similar".

Greets

File: tensorflow/python/data/ops/dataset_ops.py
Comment (MarkDaoust) [inline tensorflow/python/data/ops/dataset_ops.py]: This should be a `from_tensor_slices`.

File: tensorflow/python/data/ops/dataset_ops.py
Comment (MarkDaoust) [inline tensorflow/python/data/ops/dataset_ops.py]: Maybe say:

    For example, to implement `unbatch`:

File: tensorflow/python/data/ops/dataset_ops.py
Comment (MarkDaoust) [inline tensorflow/python/data/ops/dataset_ops.py]: It's probably easier for new users to mention `interleave` as a generalization of `flat_map`, instead of `flat_map` as a simplification of `interleave.`

**Review by MarkDaoust — CHANGES_REQUESTED** [review]

**Comment by Smokrow** [comment]
@MarkDaoust  edited the description

**Comment by MarkDaoust** [comment]
I can't merge this without @mrry's approval, so I'll leave it to him to have a look.

File: tensorflow/python/data/ops/dataset_ops.py
Comment (mrry) [inline tensorflow/python/data/ops/dataset_ops.py]: s/a similar outputs as/the same output as/

File: tensorflow/python/data/ops/dataset_ops.py
Comment (mrry) [inline tensorflow/python/data/ops/dataset_ops.py]: Thanks for adding an example. Please move it above the mention of "interleave".

I'd suggest replacing the words "to implement unbatch" with "to flatten a dataset of batches into a dataset of their elements".

**Comment by Smokrow** [comment]
@mrry Thanks for the feedback. I implemented the suggestions

**Review by mrry — DISMISSED** [review]

**Review by MarkDaoust — DISMISSED** [review]

**Pull request merged**
